main.go

```diff
@@ -43,6 +32,14 @@ func main() {
 	}
 }
 
+func ReadConfigurations(fileName string) (configuration, error) {
```

Comment:
1. Make it lowercase readConfiguration. Uppercase makes it exported to external package, that's not the case. 

2. Error should be treat inside this function